sgqlc/types/__init__.py

```diff
@@ -529,6 +523,14 @@ def __getattr__(cls, key):
         except KeyError as exc:
             raise AttributeError('%s has no field %s' % (cls, key)) from exc
 
+    def __dir__(cls):
+        original_dir = super(ContainerTypeMeta, cls).__dir__(cls)
+        try:
+            fields = list(cls._fields.keys())
+        except AttributeError as e:
```

Comment:
Likely this `AttributeError` is due the incorrect name. `cls.__fields` is set right below `super()`, but it could be moved before that if needed.